src/main/kotlin/AzureServiceBusBroker.kt

```diff
@@ -0,0 +1,109 @@
+import com.azure.messaging.servicebus.*
+import com.azure.messaging.servicebus.administration.ServiceBusAdministrationClientBuilder
+import com.zamna.kotask.*
+import kotlinx.coroutines.runBlocking
+import java.time.OffsetDateTime
+import java.util.*
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+import kotlin.time.toJavaDuration
+
+class AzureServiceBusBroker(
+    val uri: String
+): IMessageBroker {
+    val delayHeaderName = "delay"
+
+    private val manager = ServiceBusAdministrationClientBuilder()
+        .connectionString(uri)
+        .buildClient()
+
+    private val queues: MutableMap<String, ServiceBusSenderClient> = mutableMapOf()
+
+    private fun getSender(queueName: QueueName): ServiceBusSenderClient {
+        return queues.getOrPut(queueName) {
+            return ServiceBusClientBuilder()
+                .connectionString(uri)
+                .sender()
+                .queueName(queueName)
+                .buildClient()
+        }
+    }
+
+    override fun submitMessage(queueName: QueueName, message: Message) {
+        assertQueue(queueName)
+
+        val msg = ServiceBusMessage(message.body)
+            .setMessageId(UUID.randomUUID().toString())
+
+        val kotaskHeaders = message.headers
+            .mapKeys { (k, _) -> "$HEADERS_PREFIX$k" }
+            .plus(delayHeaderName to message.delayMs.toString())
```

Comment:
am I right that this delay header is not necessary for the Service Broker delay mechanism? Maybe we can omit it?

Reply:
I decided to keep to keep messages in different brokers look as similar as possible.